### PR TITLE
Pin pkg and test-infra to release-0.17 branch

### PR DESCRIPTION
**Pull request by pierDipi**
## Proposed Changes

- Pin and knative/test-infra to release-0.17.
  - In preparation for release-0.17 of knative/eventing in a week.

### CODE REVIEW

File: hack/update-deps.sh [hunk 1]
```diff
@@ -32,7 +32,7 @@ cd "${ROOT_DIR}"
 # The list of dependencies that we track at HEAD and periodically
 # float forward in this repository.
 FLOATING_DEPS=(
-  "knative.dev/test-infra@${VERSION}"
+  "knative.dev/test-infra@release-0.17"
```
Comment (slinkydeveloper) [inline hack/update-deps.sh:35]: That's how they do in eventing too?

Comment (pierDipi) [reply hack/update-deps.sh:35]: Yes https://github.com/knative/eventing/pull/3818/files#diff-58d4b0ee04af7710eefa27cf91255798R34

File: hack/update-deps.sh [hunk 2]
```diff
@@ -32,7 +32,7 @@ cd "${ROOT_DIR}"
...
   "knative.dev/eventing@${VERSION}"
```
Comment (slinkydeveloper) [inline hack/update-deps.sh:36]: I suppose we need to align this version too?

Comment (pierDipi) [reply hack/update-deps.sh:36]: Yeah, but not now, a day before our release day, `eventing` will be released and we will pin it here.

Comment (slinkydeveloper) [reply hack/update-deps.sh:36]: SGTM

**Pull request merged**